File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/06. Easter Decoration.py
count_customers = int(input())

order = None

current_total = 0
current_count_orders = 0

total = 0

for customer in range(count_customers):
    while True:
        order = input()
        if order == "Finish":
            break
        current_count_orders += 1
        if order == "basket":
            current_total += 1.5
        elif order == "wreath":
            current_total += 3.8
        elif order == "chocolate bunny":
            current_total += 7
    if current_count_orders % 2 == 0:
        current_total -= current_total * 0.2
    print(f"You purchased {current_count_orders} items for {current_total:.2f} leva.")
    total += current_total
    current_count_orders = 0
    current_total = 0

print(f"Average bill per client is: {(total / count_customers):.2f} leva.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/06. Easter Decoration.js
function easterDecoration(input) {
    // Mask
    let clients = Number(input.shift());
    
    let prices = {'basket': 1.50, 'wreath': 3.80, 'chocolate bunny': 7};
    let total = 0;
    
    for (let client = 1; client <= clients; client += 1) {
        let count = 0;
        let price = 0;
        while('commmand is Finish') {
            let product = input.shift();
            if (product == "Finish") {
                // TODO
                if (count % 2 == 0) {price *= 0.8;}
                total += price;
                console.log(`You purchased ${count} items for ${price.toFixed(2)} leva.`);
                break;
            }
            price += prices[product];
            count += 1;
        }
    }

    console.log(`Average bill per client is: ${(total / clients).toFixed(2)} leva.`);
}

easterDecoration([2, 'basket', 'wreath', 'chocolate bunny', 'Finish', 'wreath', 'chocolate bunny', 'Finish']);  
// Should return:
// You purchased 3 items for 12.30 leva.
// You purchased 2 items for 8.64 leva.
// Average bill per client is: 10.47 leva.


easterDecoration([1, 'basket', 'wreath', 'chocolate bunny', 'wreath', 'basket', 'chocolate bunny', 'Finish']);  
// Should return:
// You purchased 6 items for 19.68 leva.
// Average bill per client is: 19.68 leva.
